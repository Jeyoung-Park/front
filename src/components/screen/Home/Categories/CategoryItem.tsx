import React from 'react';
import {Text, View, StyleSheet, TouchableOpacity} from 'react-native';
import {CategoryInterface} from '../../../../types';

interface CategoryItemProps {
  currentCategoryId: number;
  setCurrentCategoryId: (param: number) => void;
  categoryItem: CategoryInterface;
}

const CategoryItem = ({
  currentCategoryId,
  setCurrentCategoryId,
  categoryItem,
}: CategoryItemProps) => {
  return (
    <TouchableOpacity
      onPress={() => setCurrentCategoryId(categoryItem.categoryId)}>
      <Text
        style={[
          styles.text,
          categoryItem.categoryId === currentCategoryId && styles.highlightText,
        ]}>
        {categoryItem.categoryName}
      </Text>
    </TouchableOpacity>
  );
};

export default CategoryItem;

const styles = StyleSheet.create({
  text: {paddingVertical: 8},
  highlightText: {
    borderBottomColor: 'gray',
    borderBottomWidth: 1,
    fontWeight: 'bold',
  },
});
