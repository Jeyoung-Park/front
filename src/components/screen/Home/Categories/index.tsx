import * as React from 'react';
import {Text, View, StyleSheet, FlatList} from 'react-native';
import {TouchableOpacity} from 'react-native-gesture-handler';
import {CategoryInterface} from '../../../../types';

interface CategoriesProps {
  categories: CategoryInterface[];
  currentCategoryId: number;
  setCurrentCategoryId: (param: number) => void;
}

const Categories = ({
  categories,
  currentCategoryId,
  setCurrentCategoryId,
}: CategoriesProps) => {
  const renderItem = ({item}: {item: CategoryInterface}) => {
    return (
      <TouchableOpacity onPress={() => setCurrentCategoryId(item.categoryId)}>
        <Text
          style={[{paddingVertical:8,}, item.categoryId === currentCategoryId && styles.highlightText]}>
          {item.categoryName}
        </Text>
      </TouchableOpacity>
    );
  };

  return (
    <FlatList
      style={styles.flatList}
      contentContainerStyle={styles.containerStyle}
      data={categories}
      keyExtractor={item => String(item.categoryId)}
      renderItem={renderItem}
    />
  );
};

export default Categories;

const styles = StyleSheet.create({
  flatList: {
    height: 50,
    width: '100%',

    marginTop: 16,
  },
  containerStyle: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  highlightText: {borderBottomColor: 'gray', borderBottomWidth: 1, fontWeight:'bold'},
});
