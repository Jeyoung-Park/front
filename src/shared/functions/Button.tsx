import * as React from 'react';
import { Text, View, StyleSheet, TouchableOpacity } from 'react-native';
import { TextInput } from 'react-native-gesture-handler';

interface ButtonProps {
	style: object;
	text: string;
	action: Function;
	opposite : boolean;
	disabled : boolean;
}

const Button = ({ style, text, action ,opposite,disabled }: ButtonProps) => {
	return (
		<TouchableOpacity
			style={[styles.container, style,{backgroundColor: !opposite ? "#60fac1" : "#FFFFFF",}]}
			onPress={()=>action()}
			disabled={disabled}
		>
			<Text style={{alignSelf:'center',color:opposite ? "#60fac1" : "#FFFFFF"}}>{text}</Text>
		</TouchableOpacity>

	);

};

export default Button;

const styles = StyleSheet.create({
	container: {
		marginHorizontal: 80,
		height: 50,
		borderRadius: 50,
		justifyContent:"center",
		borderWidth:1,
		borderColor : "#60fac1"
	},


});