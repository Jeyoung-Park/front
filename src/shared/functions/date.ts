// 날짜와 관련된 함수들입니다.

interface DateToYYMMDDInterface{
    date: Date | undefined;
    separator: string;
  }

// date 객체를 yymmdd 포맷으로 변경
// separator를 통해 원하는 형식으로 변경 가능 
export const dateToYYMMDD = ({
  date,
  separator = '',
}: DateToYYMMDDInterface) => {
  if (!date) return 'error';

  let year = date.getFullYear().toString().slice(2, 4);
  let month: number | string = date.getMonth() + 1;
  month = month >= 10 ? month : `0${month}`;
  let day: number | string = date.getDate();
  day = day >= 10 ? day : `0${day}`;

  return `${year}${separator}${month}${separator}${day}`;
};
