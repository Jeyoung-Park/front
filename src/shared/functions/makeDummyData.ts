import * as Factory from 'factory.ts';
import { RoomInterface } from '../../types';

interface CategoryInterface {
  categoryId: number;
  categoryName: string;
}

export const getDummyCategories = (iter: number) => {
  const categoryFactory = Factory.Sync.makeFactory<CategoryInterface>({
    categoryId: Factory.each(i => i),
    categoryName: '이름',
  });
  return categoryFactory.buildList(iter);
};

export const getDummyRooms=(iter:number)=>{
  const roomFactory=Factory.Sync.makeFactory<RoomInterface>({
    roomId:Factory.each(i=>i),
    platformId:Factory.each(i=>i),
    bossUserId:Factory.each(i=>i),
    roomName:'roomName',
    totalPrice:10000,
    maxUser:4,
    personalPrice:10000/4,
    recentPayment: new Date(),
  });
  return roomFactory.buildList(iter);
}