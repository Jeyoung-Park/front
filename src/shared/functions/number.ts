// 숫자와 관련된 함수들입니다.

export const numberWithComma = (number: number | undefined) => {
  if (!number) return 'error';
  let regexp = /\B(?=(\d{3})+(?!\d))/g;
  return number.toString().replace(regexp, ',');
};
