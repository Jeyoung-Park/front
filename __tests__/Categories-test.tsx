import 'react-native';
import React, {useState} from 'react';
// import renderer from 'react-test-renderer';
import {render} from '@testing-library/react-native';
import Categories from '../src/components/screen/Home/Categories';
import {getDummyCategories} from '../src/shared/functions/makeDummyData';

const TestCategories = () => {
  const [currentCategoryId, setCurrentCategoryId] = useState<number>(0);

  return (
    <Categories
      categories={getDummyCategories(5)}
      currentCategoryId={currentCategoryId}
      setCurrentCategoryId={param => setCurrentCategoryId(param)}
    />
  );
};

describe('Categories', () => {
  it('render test', () => {
      const rendered=render(<TestCategories />);
      expect(rendered).toMatchSnapshot();
  });
});
