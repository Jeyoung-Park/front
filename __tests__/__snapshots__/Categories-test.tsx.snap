// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories render test 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "space-around",
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "paddingVertical": 8,
          },
          Object {
            "borderBottomColor": "gray",
            "borderBottomWidth": 1,
            "fontWeight": "bold",
          },
        ]
      }
    >
      이름
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "paddingVertical": 8,
          },
          false,
        ]
      }
    >
      이름
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "paddingVertical": 8,
          },
          false,
        ]
      }
    >
      이름
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "paddingVertical": 8,
          },
          false,
        ]
      }
    >
      이름
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "paddingVertical": 8,
          },
          false,
        ]
      }
    >
      이름
    </Text>
  </View>
</View>
`;

exports[`Categories snapshot test 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "space-around",
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "paddingVertical": 8,
          },
          false,
        ]
      }
    >
      이름
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "paddingVertical": 8,
          },
          false,
        ]
      }
    >
      이름
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "paddingVertical": 8,
          },
          false,
        ]
      }
    >
      이름
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "paddingVertical": 8,
          },
          false,
        ]
      }
    >
      이름
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "paddingVertical": 8,
          },
          false,
        ]
      }
    >
      이름
    </Text>
  </View>
</View>
`;
